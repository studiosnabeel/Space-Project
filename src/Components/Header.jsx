import React from 'react'
import { NavLink } from 'react-router-dom'
import PlanetLogo from '../Assets/planet.png'


const Header = () => {
	return (
		
			<header className=' flex justify-between items-center p-2 px-6 md:px-12 m-auto'>
			
				<div className='flex justify-between items-center m-auto md:m-0'>
					 <img className='w-14 h-14 md:w-16 md:h-16' src={PlanetLogo} alt="Planet Logo" /> 
					 <h2 className='font-bold text-2xl md:text-3xl pl-4 '>Space Travelers' Hub</h2> </div>
				
				<nav className='hidden md:flex justify-between text-xl items-center text-sky-600 w-[400px]'>
					<NavLink className={({isActive})=> isActive ? "underline underline-offset-2" : undefined} to='/rockets'>Rockets</NavLink>
					<NavLink className={({isActive})=> isActive ? "underline underline-offset-2" : undefined} to='/dragons'>Dragons</NavLink>
					<NavLink className={({isActive})=> isActive ? "underline underline-offset-2" : undefined} to='/missions'>Missions</NavLink>
					<div className='w-0.5 h-4 bg-black'></div>
					<NavLink className={({isActive})=> isActive ? "underline underline-offset-2" : undefined} to='/profile'>My Profile</NavLink> 
				</nav>
			
			</header>	
	
	)
}

export default Header
