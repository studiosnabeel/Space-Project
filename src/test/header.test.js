import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import Header from '../Components/Header';

describe('Test header', () => {
  test('should Test header component', () => {
    const tree = render(
      <BrowserRouter>
        <Header />
      </BrowserRouter>,
    );
    expect(tree).toMatchSnapshot();
  });
});
