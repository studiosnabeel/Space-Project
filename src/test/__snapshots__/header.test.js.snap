// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test header should Test header component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class=" flex justify-between items-center p-2 px-6 py-4 md:px-12 m-auto border-b-2 "
      >
        <div
          class="flex justify-between items-center m-auto md:m-0"
        >
          <img
            alt="Planet Logo"
            class="w-14 h-14"
            src="planet.png"
          />
          <h2
            class="font-bold text-xl md:text-2xl pl-4 "
          >
            Space Travelers Hub
          </h2>
        </div>
        <nav
          class="hidden md:flex justify-between text-lg items-center text-sky-600 w-[400px]"
        >
          <a
            href="/rockets"
          >
            Rockets
          </a>
          <a
            href="/dragons"
          >
            Dragons
          </a>
          <a
            href="/missions"
          >
            Missions
          </a>
          <div
            class="w-0.5 h-4 bg-black"
          />
          <a
            href="/profile"
          >
            My Profile
          </a>
        </nav>
      </header>
    </div>
  </body>,
  "container": <div>
    <header
      class=" flex justify-between items-center p-2 px-6 py-4 md:px-12 m-auto border-b-2 "
    >
      <div
        class="flex justify-between items-center m-auto md:m-0"
      >
        <img
          alt="Planet Logo"
          class="w-14 h-14"
          src="planet.png"
        />
        <h2
          class="font-bold text-xl md:text-2xl pl-4 "
        >
          Space Travelers Hub
        </h2>
      </div>
      <nav
        class="hidden md:flex justify-between text-lg items-center text-sky-600 w-[400px]"
      >
        <a
          href="/rockets"
        >
          Rockets
        </a>
        <a
          href="/dragons"
        >
          Dragons
        </a>
        <a
          href="/missions"
        >
          Missions
        </a>
        <div
          class="w-0.5 h-4 bg-black"
        />
        <a
          href="/profile"
        >
          My Profile
        </a>
      </nav>
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
